torchx/lib/torchx.ex

```diff
@@ -1,2 +1,21 @@
 defmodule Torchx do
+  alias Torchx.NIF
+
+  @doc """
+  Check if device of the given type is available for Torchx.
+  Device atom can be any of
+  [:cpu, :cuda, :mkldnn, :opengl, :opencl, :ideep, :hip, :fpga, :msnpu, :xla, :vulkan, :metal, :xpu].
+
+  But only :cuda is supported for now.
```

Comment:
Only cuda is supported or only cuda can be checked? :)

Reply:
That's a very good question! I think it's the latter.